### PR TITLE
fix(reward): stabilize log-exp loss for large gaps

### PR DESCRIPTION
**Pull request by RerankerGuo**
## Background

`LogExpLoss` currently evaluates the pairwise objective as:

```python
torch.log(1 + torch.exp(reject_reward - chosen_reward))
```

This direct exponential overflows for large reward gaps. For example, a rejected-minus-chosen gap of `100` produces an infinite loss and a `NaN` gradient in float32, which can poison reward model training when `--model.loss_type` selects the log-exp path.

## Changes

- Evaluate the same objective with `torch.nn.functional.softplus`.
- Add a regression test that checks both the large-gap loss value and gradient finiteness.

`softplus(x)` is mathematically equivalent to `log(1 + exp(x))` while using a numerically stable implementation.

## Verification

Before:

```text
gap=100  loss=inf  grad=nan
gap=1000 loss=inf  grad=nan
```

After:

```text
gap=100  loss=100.0  grad=-1.0
gap=1000 loss=1000.0 grad=-1.0
```

Commands:

```bash
.venv/bin/python -m pytest tests/test_loss_aggregation.py -q
# 9 passed

.venv/bin/python -m pytest tests -q
# 18 passed

.venv/bin/pre-commit run --files openrlhf/models/loss.py tests/test_loss_aggregation.py
# all hooks passed
```

The full local test run used a minimal `deepspeed` import stub because DeepSpeed is not available on macOS; no DeepSpeed behavior is exercised by these tests.

## Impact

- Breaking change: No
- Affected module: reward model pairwise log-exp loss
- Performance impact: None expected
- Scope: The default sigmoid loss and other training losses are unchanged

## Checklist

- [x] The change is focused on one numerical correctness issue.
- [x] Regression coverage includes forward and backward behavior.
- [x] Existing tests pass locally.
- [x] Pre-commit hooks pass.

### CODE REVIEW

**Comment by RerankerGuo** [comment]
Revalidated this numerical-stability fix against the current `main` (`bc71bb1`):

- before: `loss=inf`, `grad=nan` for a reward gap of 1000
- after: `loss=1000.0`, `grad=-1.0`
- `tests/test_loss_aggregation.py`: 9 passed
- pre-commit on both changed files: all hooks passed
- branch remains current and mergeable

The local macOS test run used the minimal DeepSpeed import stub noted in the PR description; the regression itself exercises only the PyTorch loss path.
